earth2mip/diagnostic/time_loop.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+import datetime
+from typing import Optional, Any, Iterator, Tuple, List
+from earth2mip.diagnostic.base import DiagnosticBase
+from earth2mip.time_loop import TimeLoop
+from earth2mip.diagnostic.utils import filter_channels
+
+
+class DiagnosticTimeLoop(TimeLoop):
+    """Diagnostic Timeloop. This is an iterator that executes a list of diagnostic
+     models on top of a model Timeloop.
+
+    Note
+    ----
+    Presently, grids must be consistent between diagnostics and the model
+
+    Parameters
+    ----------
+    diagnostics : List[DiagnosticBase]
+        List of diagnostic functions to execute
+    model : TimeLoop
+        Model inferencer iterator
+    concat : bool, optional
+        Concatentate diagnostic outputs with model outputs, by default True
```

Comment:
please remove all type information from all docstrings, so that we don't need to worry about inconsistency with the type hints.